str/associatr/multiple_testing_correction/run_gene_level_pval.py

```diff
@@ -0,0 +1,306 @@
+)
+@click.option(
+    '--acat',
+    is_flag=True,
+    help='Run ACAT method',
+)
+@click.option(
+    '--bonferroni',
+    is_flag=True,
+    help='Run Bonferroni method',
+)
+@click.command()
+def main(input_dir, cell_types, chromosomes, max_parallel_jobs, acat, bonferroni):
+    """
+    Compute gene-level p-values
+    """
+    # Setup MAX concurrency by genes
+    _dependent_jobs: list[hb.batch.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.batch.job.Job):
+        """
+        To avoid having too many jobs running at once, we have to limit concurrency.
+        """
+        if len(_dependent_jobs) >= max_parallel_jobs:
+            job.depends_on(_dependent_jobs[-max_parallel_jobs])
+        _dependent_jobs.append(job)
+
+    genes_per_job = 70
+
+    for cell_type in cell_types.split(','):
+        for chromosome in chromosomes.split(','):
+            gene_files = list(
+                to_path(f'{input_dir}/{cell_type}/chr{chromosome}').glob('*.tsv')
+            )
+            for i in range(0, len(gene_files), genes_per_job):
+                batch_gene_files = gene_files[i : i + genes_per_job]
+                j = get_batch('Compute gene level pvals').new_python_job(
+                    name=f'Compute gene-level p-values for genes {i+1}-{i+genes_per_job}'
+                )
+                j.cpu(0.25).memory('lowmem')
+                f = get_batch('Compute gene level pvals').new_python_job(
+                    name=f'Compute gene-level Bonferroni p-values for genes {i+1}-{i+genes_per_job}'
+                )
+                f.cpu(0.25).memory('lowmem')
+                for gene_file in batch_gene_files:
+                    # read the raw results
+                    gene_results = pd.read_csv(gene_file, sep='\t')
+                    pvals = gene_results.iloc[:, 5]  # stored in the 6th column
+                    # Find and store the attributes of the locus with lowest raw pval
+                    # Find the minimum value in column 6
+                    min_value = gene_results.iloc[:, 5].min()
+                    # Find the rows with the minimum value in column 6
+                    min_rows = gene_results[gene_results.iloc[:, 5] == min_value]
+
+                    if len(min_rows) == 1:
+                        chr = [min_rows.iloc[:, 0].values[0]]
+                        pos = [min_rows.iloc[:, 1].values[0]]
+                        n_samples_tested = [min_rows.iloc[:, 3].values[0]]
+                        raw_pval = [min_rows.iloc[:, 5].values[0]]
+                        coeff = [min_rows.iloc[:, 6].values[0]]
+                        se = [min_rows.iloc[:, 7].values[0]]
+                        r2 = [min_rows.iloc[:, 8].values[0]]
+                        motif = [min_rows.iloc[:, 9].values[0]]
+                        ref_len = [min_rows.iloc[:, 11].values[0]]
+                        allele_frequency = [min_rows.iloc[:, 12].values[0]]
```

Comment:
In general this is a wild number of values to pass around. I'd personally replace this with something like:

```
# store a mapping of the key description to the index
VALUES_TO_INDEXES = [('chr', 0), ('pos', 1), ...]

# create a dictionary of {key: list}
row_dict = {key: [] for key, value in VALUES_TO_INDEXES}

# populate the dict
for _index, row in min_rows.iterrows():
    for key, value in VALUES_TO_INDEXES:
        row_dict[key].append(row.iloc[value])

# then pass the dict to each method as a single argument

# writing out to a line can be done tidily with 
f.write('\t'.join([dict[key] for key, value in VALUES_TO_INDEXES])
# which would auto-adjust if you add or remove target values/indexes
```

This might not work for your use case. I'm not sure the idiom "If you have a procedure with 10 parameters, you probably missed some" really applies here, but this feels like a lot of separate variables being passed around, especially for the bonferroni method that only changes one of them.